File: bob/Bob.cs
using System;
using System.Linq;

public static class Bob
{
    public static string Response(string statement)
    {
        if (statement.IsEmpty())
            return "Fine. Be that way!";
        if (statement.IsQuestion() && statement.IsYell())
            return "Calm down, I know what I'm doing!";
        if (statement.IsQuestion())
            return "Sure.";
        if (statement.IsYell())
            return "Whoa, chill out!";
        
        return "Whatever.";
    }

    private static bool IsQuestion(this string statement) =>
        statement.Trim().EndsWith("?");

    private static bool IsYell(this string statement) => 
        statement.Any(char.IsUpper) && !statement.Any(char.IsLower);

    private static bool IsEmpty(this string statement) =>
        string.IsNullOrEmpty(statement) || string.IsNullOrWhiteSpace(statement);
}

File: fsharp/bob/Bob.fs
module Bob

open System

let private isEmpty (input : string) =
    String.IsNullOrWhiteSpace(input)

let private isYell (input : string) =
    Seq.exists Char.IsLetter input && input = input.ToUpperInvariant()

let private isQuestion (input : string) =
    input.Trim().EndsWith "?"

let private fine = "Fine. Be that way!"
let private camlDown = "Calm down, I know what I'm doing!"
let private chill = "Whoa, chill out!"
let private sure = "Sure."
let private whatevs = "Whatever."

let response (input: string) : string  =
    match input with
    | i when i |> isEmpty -> fine
    | i when i |> isYell && i |> isQuestion -> camlDown
    | i when i |> isYell -> chill
    | i when i |> isQuestion -> sure 
    | _ -> whatevs